### ISSUE TITLE
No MotionDetected update for Reolink RLC-410-5MP camera

### ISSUE THREAD

**Issue by flispy** [iobroker-community-adapters/ioBroker.synology#212]
Hi,

I have 3 cameras in surveillance station. 2 of them are working fine, unfortunately the third one only partly.
in iobroker I get all information about the camera, but the object "MotionDetected" is never updated.
What steps could I perform to get it working or help you to identify if there is an issue in adapter?
One hint: I am getting notifications via iOS App DS Cam from that camera - so in general the motion sensor is working.

<img width="1188" alt="Bildschirm­foto 2022-11-14 um 09 26 51" src="https://user-images.githubusercontent.com/15381702/201611568-2e6adf02-3933-4a42-b3aa-9e824ba9080e.png">

**Comment by mcm1957** [comment]
The adapter uses the information provided by surveillance station.
The first step would be to check whether survillance station recognizes a motion. If the motion is not detected by surveillance staion the adapter could not display it.

You write you have 3 cameras. Are the all of the same type?

**Comment by flispy** [comment]
> The adapter uses the information provided by surveillance station. The first step would be to check whether survillance station recognizes a motion. If the motion is not detected by surveillance staion the adapter could not display it.
> 
> You write you have 3 cameras. Are the all of the same type?

Hi @mcm1957 , thanks for your reply. First, yes - Motion is detected in Synology Surveillance station for this camera. Second: all cameras are from different type but same vendor. Reolink RLC-410-5MP, RLC-820A and RLC-822A. But only RLC-410-5MP is not working with MotionDetection.

**Comment by mcm1957** [comment]
OK, i'll classify this as bug at this time then.
If synlogy detects to motion but the adapter does not signal it something is wrong.
The problem might be located at iobroker.synology OR at the underlying synlogy npm module.

As I do not even operate a survaillance station I cannot analyze further at this time. Mayby someone else will pick it up. Otherwise I might revie the synology adapter but not before start of next year (2023). Sorry

**Comment by SvenJ76** [comment]
Just to jump in, glad I found this issue because I thought it was something on my side. 
Same thing here with Reolink RLC-511WA and a E1 Outdoor. Motion (and person) detection works flawlessly in Surv.station, Events are triggered but the according "motiondetected" object is never updated.

**Comment by flispy** [comment]
Hi @SvenJ76 - thanks for jumping in. Hopefully it can be fixed in the next weeks.
@mcm1957 - If there is anything to test or support you with fixing this issue, please do not hesitate do ask. It would be a pleasure.

**Comment by SvenJ76** [comment]
Tried my very best to do more debugging... the motiondetected-State is set a two locations in main.js.
The function "parselistEvents" which seems to be use nowhere and "parselistCameras". ParselistCameras sets the object "motiondetected" based on the response for "recstatus" (states.SurveillanceStation.cameras[arr[i].name].motionDetected = arr[i].recStatus === 2;)  I dumped the Json-response (which is quite big) and there is no "recStatus" in there.
Don't where this comes from, but maybe it helps a bit at least.

**Comment by SvenJ76** [comment]
Did some further debugging with my limited knowledge. As it seems, the response for "recStatus" in the parselistCameras function never gives back the needed value of "2" (eq  "Motion detect recording schedule" as called in the API Reference). The return value is always "1" for "Continue recording schedule". That's sort of true for my config, I am running a continuous recording at a lower quality and highres recording when a motion event occurs. 
Don't know if this helps a bit... hope so... ;-)

**Comment by flispy** [comment]
Hi, I also did some further investigations.
`I am using web-request url http://192.168.178.11/api.cgi?cmd=GetMdState&channel=0&user=USER&password=PASSWORD`
and getting:
`[ { "cmd" : "GetMdState", "code" : 0, "value" : { "state" : 0 } } ]`
once there is motion detected, I am getting
`[ { "cmd" : "GetMdState", "code" : 0, "value" : { "state" : 1 } } ]`
So in general it should work, but does not work in this adapter here. Therefore I am wondering which request results are used for it.

**Comment by flispy** [comment]
HI @mcm1957 did you already had time to check it?

**Comment by mcm1957** [comment]
No, I did not spend time on synology adapter - and I'm not sure when this could be the case.
I only did a precheck whether this is a bug / feature request or even unrelated. 

I do not have any video system connected to synology - so I do not see how to analyze at all.
Issue will have to wait until some dev with has time for it.

**Comment by DemigodCode** [comment]
https://github.com/iobroker-community-adapters/ioBroker.synology/issues/285#issuecomment-1806943066 See here.
got it working
